app/kubernetes-ingress-controller/faq/upgrading-ingress-controller.md

```diff
@@ -0,0 +1,260 @@
+kubectl get referencegrants -A -o jsonpath='{.items[*].apiVersion}' | tr ' ' '\n' | sort | uniq -c
+```
+
+If the output contains `v1alpha2`, it means that there are `GRPCRoute`s or `ReferenceGrant`s (or both) using `v1alpha2` storage version and you need to update the manifests before upgrading.
+
+Otherwise, upgrade Gateway API and {{ site.kic_product_name }} following these steps:
+
+1. Ensure you are using Gateway API 1.1 (you can upgrade by following the steps in the section above).
+2. Ensure you are using {{ site.kic_product_name }} version 3.2 (or above).
+3. Update all your `GRPCRoute` manifests to use `v1` instead of `v1alpha2` and your `ReferenceGrant` manifests to use `v1beta1` instead of `v1alpha2`. This can be done by following the [upgrade guide from Gateway API](https://gateway-api.sigs.k8s.io/guides/?h=v1.2#v12-upgrade-notes).
+4. Install the standard or experimental channel of Gateway API 1.2.
+
+## Upgrade {{ site.kic_product_name }}
+
+Run the following command, specifying the old release name, the namespace where you've configured {{site.base_gateway}}, and the existing `values.yaml` configuration file.
+
+{% navtabs helm %}
+{% navtab "kong/ingress" %}
+```shell
+$ helm upgrade ${YOUR_RELEASE_NAME} kong/ingress \
+  --namespace ${YOUR_NAMESPACE} \
+  -f ./values.yaml
+```
+{% endnavtab %}
+{% navtab "kong/kong" %}
+```shell
+$ helm upgrade ${YOUR_RELEASE_NAME} kong/kong \
+  --namespace ${YOUR_NAMESPACE} \
+  -f ./values.yaml
+```
+{% endnavtab %}
+{% endnavtabs %}
+
+After the upgrade completes there is a brief period of time before the new resources are online. You can wait for the relevant Pod resources to cycle by watching them in your release namespace:
+
+```shell
+$ kubectl -n ${YOUR_RELEASE_NAMESPACE} get pods -w
+```
+
+Once the new pods are in a `Ready` state, the upgrade is complete.
+
+## Rollback
+
+If you run into problems during or after the upgrade, Helm provides a rollback mechanism to revert to a previous revision of the release.
+
+```shell
+$ helm rollback --namespace ${YOUR_RELEASE_NAMESPACE} ${YOUR_RELEASE_NAME}
+```
+
+You can wait for the rollback to complete by watching the relevant Pod resources:
+
+```shell
+$ kubectl -n ${YOUR_RELEASE_NAMESPACE} get pods -w
+```
+
+After a rollback, if you run into issues in production, consider using a testing environment to identify and correct these issues, or reference the [troubleshooting documentation](/index/kubernetes-ingress-controller/#troubleshooting).
+
+## Upgrade to {{ site.kic_product_name }} 3.0
+
+Upgrading from {{ site.kic_product_name }} 2.12 to 3.x+ is a major version change which contains breaking changes. To safely upgrade, follow these steps.
+
+1. **Switch to Helm as your deployment mechanism.**
+
+    As of {{ site.kic_product_name }} 3.0, [Helm](https://github.com/Kong/charts/) is the only officially supported install method.
+
+1. **Upgrade Kong to version 3.4.1 or later.**
```

Comment:
```suggestion
1. **[Upgrade {{site.base_gateway}}](/gateway/upgrade/) to version 3.4.1 or later.**
```

Reply:
The Gateway upgrade guide isn't relevant here (confusing, I know)